File: go/BinaryTree/226_Invert_Binary_Tree/226_Invert_Binary_Tree.go
package binarytree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}

	invertTree(root.Left)
	invertTree(root.Right)

	root.Left, root.Right = root.Right, root.Left
	return root

}

File: c/07_trees/226_invert_binary_tree.c
#include "../tree.h"
#include <stdlib.h>
#include <stdio.h>




struct TreeNode *invertTree(struct TreeNode* root) {

    if (root == NULL) {
        return root;
    }
    struct TreeNode* inverted_right = invertTree(root -> right);
    struct TreeNode* inverted_left = invertTree(root -> left);
    root -> right = inverted_left;
    root -> left = inverted_right;
    return root;
}


int main() {
    int array[10] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
    int arraySize = 10;
    int bufferArray[10] = {0};

    struct TreeNode* root = build_tree(array, arraySize, bufferArray);
    print_tree(root, 1);
    root = invertTree(root);
    print_tree(root, 1);

}

